include/rocksdb/advanced_options.h

```diff
@@ -483,6 +483,17 @@ struct AdvancedColumnFamilyOptions {
   // Default: false
   bool level_compaction_dynamic_level_bytes = false;
 
+  // The ratio to calcuate the total tolerant bytes of ingested files for one
+  // level.
+  // tolerant_bytes = ingest_tolerant_ratio /
+  //                  (level - base_level + 1) * level_max_bytes.
```

Comment:
This magic formula bugs me a little. It sounds reasonable at first because smaller level needs bigger ratio to have comparable `tolerant` space. But the reason for a comparable space is unclear to me. Should we try using the same ratio across level? Which seems more consistent with the old `static_level_bytes` implementation.

Reply:
Suppose with the target sizes [- - 100MB 1G 10G 100G] and tolerant_ratio = 5, the tolerant_bytes for L3 is only 500M and that of L5 is 50G. 500M is so small that it may be easily full in the process of balancing or lightning loading, whereas 50G seems too large. So make a comparable space for lower level.

Reply:
Sounds right, but consider the initial motive of this PR, which is to tolerate some ingestion until dynamic-level grows and resolve the need to compact. So ideally we are aiming for the case where ingestions are spread proportionally across levels. If 500MB is too small and 50GB is too large, I doubt the workload itself isn't a nice fit for our optimization.

Reply:
Also, I have another reason. The extra ingestion of 500MB into base level will be amplified to more than 50GB in bottommost level.

Reply:
![image](https://user-images.githubusercontent.com/13497871/99238859-37d4c200-2835-11eb-9ecc-ea6d2379734b.png)
<img width="700" alt="截屏2020-11-17 上午11 56 37" src="https://user-images.githubusercontent.com/13497871/99345002-3dc9b200-28cc-11eb-8095-a1252d9bb9f1.png">

Yeah, from metrics ingestions are spread proportionally across levels except for L0. But as you can see, base level(L3) can still accumulate a lot of bytes sometimes.

Reply:
Though I also prefer a simple formula, just afraid that the base level is too small to tolerant SSTs at a burst.

Reply:
What do you think @yiwu-arbug

Reply:
Sorry for late response. To be frank I don't think I understand this optimization very well, but if there are reasons of both of the two approach, I'm include to a simpler formula.